docs/django.rst

```diff
@@ -0,0 +1,130 @@
+template. This time the tree is walked *within the template*, such that each
+Node is converted into an appropriate chunk of HTML and concatenated with its
+siblings. Perhaps confusingly, templates are passed the Node data as a
+"skeleton" of a full regulation -- the single section (or whatever component
+we care about) is "wrapped" with empty Nodes until it looks like a full
+regulation. This means that, from the template perspective, there is largely
+only *one* entry point for views, regardless of whether that view is
+generating a section, a single paragraph, or an entire regulation. The
+practice no doubt stems from the original, full-regulation-generation
+functionality.
+
+  There's a tremendous amount of refactoring that should happen here. We
+  shouldn't be walking the tree twice (once within ``HTMLBuilder`` and once
+  within the templates) -- it'd make more sense to remove the former altogether.
+  Further, a conversion from the ``dict`` to a class would seem appropriate, to
+  make it obvious where to look for functionality. Though the skeleton concept
+  has merit, the hoops it causes us to jump through are rather strange. Perhaps
+  a better solution would be to select an appropriate template automatically
+  based on its type, position in the tree, etc.
+
+Views
+-----
+
+There are three primary categories for our views: "sidebar", "partial", and
+"chrome". The first two stem from our AJAX needs; for browsers with the
+capability, we AJAX load in content as the user clicks around. The "partial"
+endpoints correspond to the center content of the page (e.g. a regulation
+section, search results, the diff view, etc.). When a user clicks to load a
+new section, their browser will make two AJAX requests, one for the center
+content and one for the sidebar content.
+
+The "chrome" endpoints wrap these two other types of views with navigation,
+CSS includes, headers, etc. (i.e. the application's "chrome"). These endpoints
+are crucial for users without Javascript (or modern implementations of the URL
+push API) and for initial loading (e.g. via hard refreshes, bookmarks, etc.).
+
+  We currently have far too many *different* views, despite them performing
+  largely the same types of tasks. It would make more sense to combine all of
+  the "node" views into a single class. Similarly, we *mirror* each "partial"
+  view class with a "chrome" class; a more effective strategy would be to have
+  a more generic ``wrap_with_chrome`` method and no distinct "chrome" classes.
+  This should also remove the incredibly nasty manipulations of Django's
+  request/response life cycle we're currently performing to populate the
+  chrome version. Somewhat related, having a separate endpoint for the sidebar
+  and a separate endpoint for partials didn't turn out as useful as we
+  expected.  It probably makes sense to combine them again.
+
+Layers
+------
+
+We have a handful of layer generating classes, which know how to apply data
```

Comment:
Having more examples of layers would be helpful. Also, having examples of each layer category, with how one would choose inline vs search-and-replace will make the categories more clear.